### PR TITLE
[MRG] BUG: read_spikes wasn't returning updated CellResponse instance

### PR DESCRIPTION
**Pull request by rythorpe**
Since this is an obvious 1-line fix that allows us to read legacy spk.txt files, unless there are any objections, I don't think we need to add a test here.

### CODE REVIEW

**Comment by ntolley** [comment]
Thanks @rythorpe! Do you think it's possible to add a test that ensures types are updated?

**Comment by jasmainak** [comment]
agreed, test would be nice! Also don't forget to update `whats_new` so we document your contribution :)

File: hnn_core/tests/test_cell_response.py [hunk 1]
```diff
@@ -146,11 +146,16 @@ def test_cell_response(tmpdir):
...
-    # Check for gid_ranges errors
-
     for fname in sorted(glob(str(tmpdir.join('spk_*.txt')))):
         times_gids_only = np.loadtxt(fname, dtype=str)[:, (0, 1)]
         np.savetxt(fname, times_gids_only, delimiter='\t', fmt='%s')
+
+    # Check that spike_types are updated according to gid_ranges
+    cell_response = read_spikes(tmpdir.join('spk_*.txt'),
+                                gid_ranges=gid_ranges)
+    assert cell_response.spike_types == spike_types
```
Comment (rythorpe) [inline hnn_core/tests/test_cell_response.py:156]: Is this what you were thinking @ntolley, a round-trip test to make sure that spike types can be recovered from legacy spike files given that the user passes in `gid_ranges`?

Comment (ntolley) [reply hnn_core/tests/test_cell_response.py:156]: In this style yes, but this test still passes without the bug fix? (At least when I just tried it)

I think you'll need to create a situation where `gid_ranges` defines a cell types that never spikes, I'm pretty that's what the if statement is addressing.

File: hnn_core/cell_response.py [hunk 2]
```diff
@@ -456,5 +456,4 @@ def read_spikes(fname, gid_ranges=None):
     if gid_ranges is not None:
```
Comment (ntolley) [inline hnn_core/cell_response.py:456]:
```suggestion
    # Handle edge case where cell never spikes.
    # Otherwise it would not appear when written to a file.
    if gid_ranges is not None:
```

File: hnn_core/tests/test_cell_response.py (hunk 1 above)
Comment (rythorpe) [reply hnn_core/tests/test_cell_response.py:156]: The issue I encountered when trying to read in a legacy spike file didn't have anything to do with whether or not all the cell types spiked. Also, I just tried this test on master and it failed.

File: hnn_core/cell_response.py (hunk 2 above)
Comment (rythorpe) [reply hnn_core/cell_response.py:456]: This doesn't have to do with spiking. `gid_ranges=None` by default unless the user is required to define it when loading a legacy spike file that doesn't contain the 3rd column.

File: hnn_core/tests/test_cell_response.py (hunk 1 above)
Comment (rythorpe) [reply hnn_core/tests/test_cell_response.py:156]: Pretty much all this test does is check if `CellResponse.update_types()` has been called in the fringe case where the user defines `gid_ranges`. The logic within `read_spikes()` checks for this, however, sometime in the last few months the function was accidentally set to return the wrong `CellResponse` object.

Comment (jasmainak) [reply hnn_core/tests/test_cell_response.py:156]: I think this test is fine but it might help to move it after L28 so it's clear that the `spike_types` correspond to the `gid_ranges` provided

**Review by jasmainak — APPROVED** [review]
@ntolley I let you merge when you're happy with the PR

File: hnn_core/tests/test_cell_response.py (hunk 1 above)
Comment (rythorpe) [reply hnn_core/tests/test_cell_response.py:156]: @jasmainak this is super minor, but we'd also need to move L149-151 up as well, which would separate them from L158-167 and create a similar problem. Should I just make another copy of `gid_ranges` down so that it's easy to read even though redundant?

File: hnn_core/cell_response.py (hunk 2 above)
Comment (ntolley) [reply hnn_core/cell_response.py:456]: Ah I see what you mean, I misunderstood the situation where this is raised

File: hnn_core/tests/test_cell_response.py (hunk 1 above)
Comment (ntolley) [reply hnn_core/tests/test_cell_response.py:156]: Ah I get it now! Now I just need to figure out what's going on with my local tests...

**Review by ntolley — APPROVED** [review]
Looks great! I'll merge once you decide on where to put the new tests

File: hnn_core/tests/test_cell_response.py (hunk 1 above)
Comment (jasmainak) [reply hnn_core/tests/test_cell_response.py:156]: oh I see, let's leave it as it is then.

**Comment by rythorpe** [comment]
> Looks great! I'll merge once you decide on where to put the new tests

Go for it @ntolley!
